### PR TITLE
Reflect device name when editing message from different device

### PR DESCRIPTION
**Pull request by zanderz**

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -1213,6 +1219,11 @@ func (h *Server) PostEditNonblock(ctx context.Context, arg chat1.PostEditNonbloc
 	parg.IdentifyBehavior = arg.IdentifyBehavior
 	parg.OutboxID = arg.OutboxID
 	parg.Msg.ClientHeader.MessageType = chat1.MessageType_EDIT
+	// Sending device may have changed
+	parg.Msg.ClientHeader.SenderDevice, err = h.getGregorDeviceID()
```
Comment (mmaxim) [inline go/chat/server.go]: Can just delete all the setting of device ID here, I think this code is old and we set this later on.

File: go/chat/supersedes.go [hunk 2]
```diff
@@ -70,6 +70,12 @@ func (t *basicSupersedesTransform) transformEdit(msg chat1.MessageUnboxed, super
 	mvalid.AtMentionUsernames = superMsg.Valid().AtMentionUsernames
 	mvalid.ChannelMention = superMsg.Valid().ChannelMention
 	mvalid.ChannelNameMentions = superMsg.Valid().ChannelNameMentions
+	if superMsg.Valid().SenderDeviceName != "" {
```
Comment (mmaxim) [inline go/chat/supersedes.go]: What is the point of this blank check?

Comment (zanderz) [reply go/chat/supersedes.go]: Because there will be older clients that send superseding messages without a device name?

Comment (mmaxim) [reply go/chat/supersedes.go]: Don't think so, that is set later on in the pipeline. See my first comment, setting it there was just old code that wasn't needed.

File: go/chat/server.go (hunk 1 above)
Comment (zanderz) [reply go/chat/server.go]: for a superseding message? I don't think so but I will try it

Comment (zanderz) [reply go/chat/server.go]: ok it works without that, thanks

File: go/chat/supersedes.go (hunk 2 above)
Comment (zanderz) [reply go/chat/supersedes.go]: ok will remove

**Comment by zanderz** [comment]
@mmaxim how is it now?

File: go/chat/server.go [hunk 3]
```diff
@@ -1126,6 +1126,15 @@ func (h *Server) SetConversationStatusLocal(ctx context.Context, arg chat1.SetCo
 	}, nil
 }
 
+func (h *Server) getGregorDeviceID() (gregor1.DeviceID, error) {
```
Comment (zanderz) [inline go/chat/server.go]: This is now used in just one place - could revert this part if you want

**Comment by zanderz** [comment]
@mmaxim I think this answers your questions, how does it look now?

**Comment by mmaxim** [comment]
I just pushed up a commit which just remove some of the unneeded code, should also still fix your case.

**Pull request merged**